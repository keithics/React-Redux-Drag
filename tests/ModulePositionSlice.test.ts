import {
  ghostImageInitialState,
  GhostImageReducer,
  setGhostImageCoordinates,
} from '../src/redux/slices/GhostImageSlice';

describe('Ghost Image Tests', () => {
  test('should return the initial state', () => {
    // @ts-ignore
    expect(GhostImageReducer(undefined, {})).toMatchSnapshot();
  });

  test('should be able to change the ghost image coordinates', () => {
    expect(
      GhostImageReducer(
        ghostImageInitialState,
        setGhostImageCoordinates({
          id: 2,
          coord: {
            x: 1,
            y: 2,
            h: 100,
            w: 2,
          },
        })
      )
    ).toMatchSnapshot();
  });
});
