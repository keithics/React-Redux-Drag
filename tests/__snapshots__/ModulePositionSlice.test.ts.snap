// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ghost Image Tests should be able to change the ghost image coordinates 1`] = `
Object {
  "h": 100,
  "isCentered": true,
  "w": 159,
  "x": 94.5,
  "y": 12,
}
`;

exports[`Ghost Image Tests should return the initial state 1`] = `
Object {
  "h": 0,
  "isCentered": true,
  "w": 0,
  "x": 0,
  "y": 0,
}
`;
